compiler/z/env/OMRCPU.hpp

```diff
@@ -45,53 +47,186 @@ namespace Z
 
 class CPU : public OMR::CPU
    {
-protected:
+   public:
 
-   CPU() :
-         OMR::CPU(),
-      _s390MachineType(TR_UNDEFINED_S390_MACHINE)
-      {}
-
-public:
-   bool getSupportsHardwareSQRT() { return true; }
-
-   bool getS390SupportsZ900() { return true; }
-
-   bool getS390SupportsZ990() { return true; }
-
-   bool getS390SupportsZ9() { return true; }
-
-   bool getS390SupportsZ10() { return true; }
-
-   bool getS390SupportsZ196() { return true; }
-
-   bool getS390SupportsZEC12() { return false; }
-
-   bool getS390SupportsZ13() { return false; }
-
-   bool getS390SupportsZ14() { return false; }
-
-   bool getS390SupportsZ15() { return false; }
-
-   bool getS390SupportsZNext() { return false; }
-
-   bool getS390SupportsHPR() { return false; }
-
-   bool getS390SupportsDFP() { return false; }
-
-   bool getS390SupportsFPE() {return false; }
+   enum Architecture
+      {
+      TR_UnknownArchitecture = 0,
+      TR_z900 = 1,
+      TR_z990 = 2,
+      TR_z9 = 3,
+      TR_z10 = 4,
+      TR_z196 = 5,
+      TR_zEC12 = 6,
+      TR_z13 = 7,
+      TR_z14 = 8,
+      TR_z15 = 9,
+      TR_zNext = 10,
```

Comment:
Since this enum now lives in a class in the `TR::` namespace, I find writing `getSupportsArch(TR::CPU::TR_z10)` a little cumbersome.  Would you consider dropping the `TR_` prefix from these enum values?  It doesn't have to be fixed in this PR as it would be a broad search and replace across two projects.

Reply:
No better time than now. I fully agree!

Reply:
Fixed in 2cb3fbd